From edfeae583df511e7ba3a3cdafb7348eaf7c94796 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 18 Jul 2018 18:38:26 +0800
Subject: [PATCH 4269/5242] MLK-19017-1 ARM64: dts: imx8mm: re-order display
 nodes

commit  bf17270beed7befb0b1598995e68dc9b215241da from
https://source.codeaurora.org/external/imx/linux-imx.git

The devices in the device tree are registered in the
top-down order. But when system suspend entered, the
device tree is walked in a bottom-up order to suspend
devices. So change the display device nodes register
order to be:

    LCDIF -> SEC DSIM -> Display Subsystem

Since in display subystem, it will disable the whole
display pipeline. So this should be first suspended
before LCDIF and SEC DSIM. And besides, the SEC DSIM
is better to be suspended before LCDIF which is the
same with the sequence for display pipeline disables.
And when system resume entered, the devices resume
sequence is:

    LCDIF -> SEC DSIM -> Display Subsystem

Which is a top-down order and is the correct sequence
for display devices resume.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 4ce74783e6a50898a433372f5968175b20d4ef0c)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index 4fb1b97..eec49a6 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -1003,11 +1003,6 @@
 		};
 	};
 
-	display-subsystem {
-		compatible = "fsl,imx-display-subsystem";
-		ports = <&lcdif_disp0>;
-	};
-
 	mipi_dsi: mipi_dsi@32E10000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -1033,6 +1028,11 @@
 		};
 	};
 
+	display-subsystem {
+		compatible = "fsl,imx-display-subsystem";
+		ports = <&lcdif_disp0>;
+	};
+
 	pcie0: pcie@0x33800000 {
 		compatible = "fsl,imx8mm-pcie", "snps,dw-pcie";
 		reg = <0x0 0x33800000 0x0 0x400000>, <0x0 0x32f00000 0x0 0x10000>,
-- 
1.7.9.5

